---
layout: default
title: Teaching - 2014
---

<a name="2014"><table width="100%" bgcolor="#e6e6e6"> <tr><th style="font-variant:small-caps; font-size:12pt; color:#003399;" align="left">2014</th></tr></table></a><br>

<!-- ########################################################################################### -->

<a name="2014AdvSeq"><h3>CSHL Advanced Sequencing Course</h3></a> <hr>

More information is available on the course <a href="http://meetings.cshl.edu/courses/2014/c-seqtec14.shtml">Webpage</a> <br>

<br><br>

<a href="AdvSeqCourse//Schatz-Assembly Theory and Practice.pdf">1. De novo assembly theory and practice</a><p>Theory and practice of assembly projects examining requirements for coverage, read lengths, and quality with a focus on Illumina and PacBio sequecning</p>
<a href="AdvSeqCourse//Schatz-Assembly Exercise.pdf">2. De novo assembly exercise</a><p>A hands on exercise assembling a genome and searching for a secret message encoded along a novel insertion</p>
<a href="AdvSeqCourse//Schatz-Genome Annotation.pdf">3. Genome Annotations</a><p>Strategies for annotating a genome, including gene prediction, alignment techniques, and high throughput functional sequencing assays</p>
<a href="AdvSeqCourse//Gurtowski.Long Reads.pdf">4. Long Read Assembly</a> <p>Latest advances in long read assembly. Presented by James Gurtowksi </p>
<a href="AdvSeqCourse//Garvin.Single_Cell_Analysis.pdf">5. Single Cell Analysis</a> <p> Latest advances in single cell analysis. Presented by Tyler Garvin</p>

<br><br>

<!-- ########################################################################################### -->

<a name="2014WSBS"><h3>CSHL WSBS Genomics and Quantitive Biology</h3></a> <hr>

<a href="Schatzlab Research Projects.pdf">Schatz Lab Research Projects</a> <p>Overview of my background and research</p>

<br><br>

<h3>QB Bootcamp</h3>

<a href="QB Bootcamp 1. Intro.pdf">QB Bootcamp 1: Introduction </a> <p>The challenges of biological data science</p>
<a href="QB Bootcamp 2. Unix Scripting.pdf">QB Bootcamp 2: Unix scripting</a> <p>A gentle introduction to working at the command line</p>
<a href="QB Bootcamp 3. ORC Exercise.pdf">QB Bootcamp 3: ORC Exercises</a> <p>Use sequencing to discover origins of replication</p>
<a href="qb1.zip">qb1.zip</a> <p>Python programming exercises</p>
<a href="qb2.zip">qb2.zip</a> <p>Shell programming exercises</p>

<br><br>

<h3>Genomics</h3>

<a href="Genomics Lecture 1. Omics Bootcamp.pdf">Genomics 1: Omics Bootcamp</a> <p>Introduction to whole genome, exome sequencing, RNA-seq, ChIP-seq, Methy-seq, and single cell analysis </p>
<a href="Genomics Lecture 2. ENCODE Discussion.pdf">Genomics 2: ENCODE</a> <p>History and major findings of ENCODE</p>
<a href="Genomics Lecture 3. Ancient and Modern Humans.pdf">Genomics 3: Ancient and Modern Humans</a> <p>Major results from 1000 genomes project, Neanderthal sequencing, surname inference</p>

<br><br>

<h3>Quantitative Biology</h3>

<a href="QB Lecture 1. Exact Matching.pdf">QB Lecture 1. Exact Matching</a> <p>Introduction to brute force, binary search, suffix arrays and the BWT</p>
<a href="QB Lecture 1. BWT Notes.pdf">QB Lecture 1: BWT Notes</a> <p>Lecture Notes on the BWT</p>
<a href="QB Lecture 2. Dynamic Programming.pdf">QB Lecture 2: Dynamic Programming</a> <p>Fibonnaci Numbers, Longest Increasing Subsequence, and Sequence Alignment</p>
<a href="QB Lecture 2. Dynamic Programming Notes.pdf">QB Lecture 2: Dynamic Programming Notes</a> <p>Lecture notes on designing a dynamic programming algorithm</p>
<a href="QB Lecture 3. Graphs and Genomes.pdf">QB Lecture 3: Graphs and Genomes</a> <p>Basic graph algorithms, methods for genome assembly</p>
<a href="QB Lecture 4. Gene Finding and HMMs.pdf">QB Lecture 4: Gene Finding and HMMs</a> <p>Microbial and Eukaryotic Gene Finding, Markov Models, HMMs and GHMMs, Forward Algorithm, Viterbi</p>

<br>
<br>


<!-- ########################################################################################### -->

<a name="2014Plants"><h3>CSHL Frontiers and Techniques in Plant Science</h3></a> <hr>

<a href="2014.07.06.PlantAssembly.pdf">Genome Sequencing and Assembly</a>
<p>
Introduction to de Bruijn graphs for genome assembly
</p>

<a href="2014.07.06.PlantAssemblyTutorial.pdf">Assembly Tutorial</a>
<p>
Assembly tutorial to detect a secret message embedded into a microbial genome
</p>

<br>
<br>



<!-- ########################################################################################### -->

<a name="2014URP"><h3>CSHL Undergraduate Research Program in Bioinformatics</h3></a> <hr>

<a href="2014.URP.Searching for GATTACA.pdf">Searching for GATTACA</a><br>
<p>
In this class we explored the problem of finding exact occurrences of a query
sequence in a large genome or database of sequences.  Under this theme, we
started by analyzing the brute force approach introducing the concepts of
algorithm, complexity analysis, and E-values.  Next we discussed suffix arrays
as an index for accelerating the search, including analyzing the performance of
binary search. We also considered two traditional algorithms for sorting
(Selection Sort versus QuickSort) and their relative performance.  In the second
half of the class we discussed finding approximate occurrences of a short query
sequence in a large genome or database of sequences. We first defined the
problem by considering various metrics of an approximate occurrence such as
hamming distance, or edit distance.  We then considered different methods for
computing inexact alignments including brute force global &amp; local
alignments, and seed-and-extend algorithms. Finally we discussed Bowtie as a
Burrows-Wheeler transform based short read mapping algorithm for discovering
alignments to reference genome.
</p>

<br>

<h3>Python &amp; Bioinformatics</h3>

<a href="python1.zip">Python Class 1</a><br>
<p>
Introduction to python, variables, lists, conditions, loops
</p>

<br>

<a href="python2.zip">Python Class 2</a><br>
<p>
Brute force search, dictionaries, motif finding
</p>

<br>

<h3>iPython Notebooks for Probability &amp; Statistics</h3>
<ol>
  <li> <a href="http://nbviewer.ipython.org/url/schatzlab.cshl.edu/teaching/2014/urp/1.Rolling%20a%20die.ipynb">Rolling a die (Uniform Random Probability)</a></li>
  <li> <a href="http://nbviewer.ipython.org/url/schatzlab.cshl.edu/teaching/2014/urp/2.Flipping%20a%20coin.ipynb">Flipping a coin (Binomial &amp; Normal Distributions)</a></li>
  <li> <a href="http://nbviewer.ipython.org/url/schatzlab.cshl.edu/teaching/2014/urp/3.Throwing%20Marbles%20into%20Jars.ipynb">Throwing Marbles into Jars (Poisson Distribution)</a></li>
  <li> <a href="http://nbviewer.ipython.org/url/schatzlab.cshl.edu/teaching/2014/urp/4.Throwing%20darts.ipynb">Throwing Darts (Exponential Distribution)</a></li>
</ol>
<br>

<p>
We also used the exercises at <a href="http://rosalind.info/problems/locations/">Rosalind</a> throughout the course.
</p>

<br>


<h3>Special topics</h3>
<p>
Talk by <a href="http://churchlandlab.cshl.edu/">Anne Churchland</a> on <a href="CareerDevelopmentWorkLifeBalanceTalk.pdf">balancing work and life.</a>
</p>

